versioned_docs/version-2.7/reference-guides/best-practices/rancher-server/tuning-and-best-practices-for-rancher-at-scale.md

```diff
@@ -21,6 +21,12 @@ This guide describes the best practices and tuning approaches to scale Rancher s
 
 When scaling up Rancher, one typical bottleneck is resource growth in the upstream (local) Kubernetes cluster. The upstream cluster contains information for all downstream clusters. Many operations that apply to downstream clusters create new objects in the upstream cluster and require computation from handlers running in the upstream cluster.
 
+### Avoiding third-party applications
+
+Running workloads or microservices in in the upstream cluster other than Rancher can cause performance issues, especially if the application makes heavy use of CRDs or otherwise creates significant load on Kubernetes components such as `etcd` or `kubeapiserver`.
```

Comment:
```suggestion
Running workloads or microservices other than Rancher in the upstream cluster can cause performance issues, especially if the application makes heavy use of CRDs or otherwise creates significant load on Kubernetes components such as `etcd` or `kubeapiserver`.
```